pkg/cloudclient/aws/aws.go

```diff
@@ -0,0 +1,212 @@
+	region := defaultRegion
+	if cd.Spec.Platform.AWS != nil {
+		// overwrite default region if it's set
+		glog.V(8).Infof("Overwriting default region with %s", cd.Spec.Platform.AWS.Region)
+		region = cd.Spec.Platform.AWS.Region
+	}
+
+	accessKeyID, secretAccessKey, err := getAWSCredentialsFromDirectory(credentialDir)
+	if err != nil {
+		glog.Errorf("Couldn't obtain credentials from secret. Error: %s", err.Error())
+		return nil, &ccerrors.InternalError{Err: err}
+	}
+
+	awsClient, err := singletonClient.getAwsClientBuilder().New(accessKeyID, secretAccessKey, "", region)
+	if err != nil {
+		glog.Errorf("Couldn't create AWS cloud client: %s", err.Error())
+		return nil, &ccerrors.UpstreamError{Err: err}
+	}
+
+	return &CloudClient{
+		clusterDeployment: cd,
+		awsClient:         awsClient,
+		ocmClient:         ocmClient,
+		region:            region,
+	}, nil
+}
+
+// The singleton client defined here is used to build the initial cloud client for a request.
+// the biggest use for this client is to be able to inject the awsclient builder during unit testing
+var singletonClient = &CloudClient{}
+
+func (c *CloudClient) getAwsClientBuilder() awsclient.BuilderIface {
+	if c.awsClientBuilder != nil {
+		return c.awsClientBuilder
+	}
+	return &awsclient.Builder{}
+}
+
+func getAWSCredentialsFromDirectory(dir string) (string, string, error) {
+	accessKeyIDFilename := "aws_access_key_id"
+	secretAccessKeyIDFilename := "aws_secret_access_key"
+
+	// ensure that the directory has no trailing slash
+	filepath := strings.TrimSuffix(dir, "/")
+
+	accessKeyBytes, err := ioutil.ReadFile(path.Join(filepath, accessKeyIDFilename))
+	if err != nil {
+		glog.Error(err)
+		return "", "", fmt.Errorf("%s cannot be read from %s", accessKeyIDFilename, filepath)
+	}
+	secretKeyBytes, err := ioutil.ReadFile(path.Join(filepath, secretAccessKeyIDFilename))
+	if err != nil {
+		glog.Error(err)
+		return "", "", fmt.Errorf("%s cannot be read from %s", secretAccessKeyIDFilename, filepath)
+	}
+	accessKeyID := strings.TrimRight(string(accessKeyBytes), "\n")
+	secretKeyID := strings.TrimRight(string(secretKeyBytes), "\n")
+	return accessKeyID, secretKeyID, nil
+}
+
+// extracts a cluster ID from a given clusterdeployment
+func extractClusterID(cd *hivev1.ClusterDeployment) (string, error) {
+	clusterID := cd.ObjectMeta.Labels[cdClusterIDLabelKey]
+	if clusterID == "" {
+		return "", &ccerrors.ValidationError{Err: fmt.Errorf("Unable to parse ClusterID from ClusterDeployment, label missing or malformed")}
+
+	}
+	return clusterID, nil
+}
+
+// abstracted functionality in order to do the assume-role chaining in order to assume the support role
+// credentials in a given customer cluster. This function will get the accountclaim, assume the jump role,
+// and then from that assumed role make another assume_role call into the customer's account
+func (c *CloudClient) AssumeSupportRole() (*sts.Credentials, error) {
+
```

Comment:
Do we have to return the STS type back?
I know it's double converting but still asking

Reply:
I think it's ok for now since the function itself AssumeSupportRole is very STS specific. 
Later on, when we may want to add more clients for other cloud providers, we can try to unify the GetCredential returns.